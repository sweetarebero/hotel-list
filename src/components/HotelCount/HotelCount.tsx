import * as React from "react";

interface Props {
  count: number;
  city: string;
}

const HotelCount = ({ count, city }: Props) => {
  return (
    <p>
      <b>{count}</b> <i> hotels in</i> <b>{city}</b>
    </p>
  );
};

export default HotelCount;
