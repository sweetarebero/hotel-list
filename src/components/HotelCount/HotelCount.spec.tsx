import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import HotelCount from "./HotelCount";

describe("HotelCount", () => {
  it("renders correctly", () => {
    const { container } = render(<HotelCount count={5} city="Sydney" />);
    expect(container).toMatchSnapshot();

    const countText = screen.getByText(/5/i);
    const countryText = screen.getByText(/Sydney/i);
    expect(countText).toBeInTheDocument();
    expect(countryText).toBeInTheDocument();
  });
});
