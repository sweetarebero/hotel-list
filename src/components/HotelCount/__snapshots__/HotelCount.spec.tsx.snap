// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelCount renders correctly 1`] = `
<div>
  <p>
    <b>
      5
    </b>
     
    <i>
       hotels in
    </i>
     
    <b>
      Sydney
    </b>
  </p>
</div>
`;
